File: cookbooks/zsh/source.rb
include_recipe '../def'

version = '5.6.2'
src_dir = '/usr/local/src'
zsh_dir = "#{src_dir}/zsh-#{version}"

package 'ncurses-dev'

curl 'download zsh archive' do
  path '/tmp/zsh.tar.xz'
  url "https://sourceforge.net/projects/zsh/files/zsh/#{version}/zsh-#{version}.tar.xz/download"
end

execute 'tar zsh' do
  cwd src_dir
  command <<-EOT
    tar xvf /tmp/zsh.tar.xz
  EOT
  not_if "test -e #{zsh_dir}"
end

execute 'configure' do
  cwd zsh_dir
  command './configure'
  not_if 'test -e config.log'
end

execute 'make and make install' do
  cwd zsh_dir
  command 'make && make install'
  not_if 'test -e /usr/local/bin/zsh'
end

execute 'add to /etc/shells' do
  cwd zsh_dir
  command <<-EOT
    echo '/usr/local/bin/zsh' >> /etc/shells
  EOT
  not_if 'grep /usr/local/bin/zsh /etc/shells'
end

File: cookbooks/git/source.clj
#!/usr/bin/env dad --no-color --dry-run

(def version "2.24.1")
(def prefix "/usr/local")
(def src-dir (str prefix "/src"))
(def git-dir-name (str "git-" version))

(package ["make"
          "libssl-dev"
          "libghc-zlib-dev"
          "libcurl4-gnutls-dev"
          "libexpat1-dev"
          "gettext"
          "unzip"])

(package "git" {:action :uninstall})

(download {:path (str src-dir "/git.zip")
           :url (str"https://github.com/git/git/archive/v" version ".zip")})

(execute {:cwd src-dir
          :command "unzip git.zip"
          :pre-not (str "test -d " git-dir-name)})

(execute {:cwd (str src-dir "/" git-dir-name)
          :command [(str "make prefix=" prefix " all")
                    (str "make prefix=" prefix " install")]
          :pre-not (str "test -e " prefix "/bin/git")})
